server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -207,6 +261,67 @@ public Iterator<Setting<?>> settings() {
 
     }
 
+    /**
+     * Validates that low, high and flood stage max headrooms adhere to the comparison: flood &lt; high &lt; low. Else, throws an exception.
+     */
+    static class MaxHeadroomValidator implements Setting.Validator<ByteSizeValue> {
+
+        @Override
+        public void validate(ByteSizeValue value) {
+
+        }
+
+        @Override
+        public void validate(final ByteSizeValue value, final Map<Setting<?>, Object> settings) {
+            final ByteSizeValue lowHeadroom = (ByteSizeValue) settings.get(CLUSTER_ROUTING_ALLOCATION_LOW_DISK_MAX_HEADROOM_SETTING);
+            final ByteSizeValue highHeadroom = (ByteSizeValue) settings.get(CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING);
+            final ByteSizeValue floodHeadroom = (ByteSizeValue) settings.get(
+                CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_MAX_HEADROOM_SETTING
+            );
+
+            // For the comparisons, we need to mind that headroom values can default to -1.
```

Comment:
Does it make sense to accept `-1` for these settings? How is that different from `0`?

Reply:
Forget this, the difference is clear with a corrected understanding of the change.